src/rider/main/kotlin/com/jetbrains/rider/plugins/efcore/startup/EfCoreStartupListener.kt

```diff
@@ -9,24 +9,32 @@ import com.jetbrains.rd.util.lifetime.Lifetime
 import com.jetbrains.rider.plugins.efcore.EfCoreUiBundle
 import com.jetbrains.rider.plugins.efcore.KnownNotificationGroups
 import com.jetbrains.rider.plugins.efcore.features.eftools.InstallDotnetEfAction
+import com.jetbrains.rider.plugins.efcore.features.eftools.SuppressEfCoreToolsInstallation
 import com.jetbrains.rider.plugins.efcore.rd.RiderEfCoreModel
+import com.jetbrains.rider.plugins.efcore.state.CommonOptionsStateService
 
 @Suppress("UnstableApiUsage")
 class EfCoreStartupListener : SolutionExtListener<RiderEfCoreModel> {
+  private val commonOptionsStateService by lazy { CommonOptionsStateService.getInstance() }
   override fun extensionCreated(lifetime: Lifetime, session: ClientProjectSession, model: RiderEfCoreModel) {
-    model.onMissingEfCoreToolsDetected.set { _, unit ->
-      NotificationGroupManager
-        .getInstance()
-        .getNotificationGroup(KnownNotificationGroups.efCore)
-        .createNotification(
-          EfCoreUiBundle.message("notification.title.ef.core.tools.required"),
-          EfCoreUiBundle.message("notification.content.ef.core.tools.are.required.to.execute.this.action"),
-          NotificationType.WARNING
-        )
-        .addAction(InstallDotnetEfAction())
-        .notify(session.project)
+    if(!commonOptionsStateService.getProjectToolsInstallationSupressed()) {
+      model.onMissingEfCoreToolsDetected.set { _, unit ->
+        NotificationGroupManager
+                .getInstance()
+                .getNotificationGroup(KnownNotificationGroups.efCore)
+                .createNotification(
+                        EfCoreUiBundle.message("notification.title.ef.core.tools.required"),
+                        EfCoreUiBundle.message("notification.content.ef.core.tools.are.required.to.execute.this.action"),
+                        NotificationType.WARNING
+                )
+                .addAction(InstallDotnetEfAction())
+                .apply{
+                  addAction(SuppressEfCoreToolsInstallation(this))
+                }
+                .notify(session.project)
```

Comment:
Please review whitespaces, don't change them

```suggestion
        NotificationGroupManager
          .getInstance()
          .getNotificationGroup(KnownNotificationGroups.efCore)
          .createNotification(
            EfCoreUiBundle.message("notification.title.ef.core.tools.required"),
            EfCoreUiBundle.message("notification.content.ef.core.tools.are.required.to.execute.this.action"),
            NotificationType.WARNING
          )
          .addAction(InstallDotnetEfAction())
          .apply{
            addAction(SuppressEfCoreToolsInstallation(this))
          }
          .notify(session.project)
```